test/parallel/test-assert.js

```diff
@@ -655,40 +655,46 @@ try {
                      'Message incorrectly marked as generated');
 }
 
-// Verify that throws() and doesNotThrow() throw on non-function block
-function testBlockTypeError(method, block) {
-  let threw = true;
-
-  try {
-    method(block);
-    threw = false;
-  } catch (e) {
-    assert.strictEqual(e.toString(),
-                       'TypeError: "block" argument must be a function');
-  }
+{
+  // Verify that throws() and doesNotThrow() throw on non-function block
+  const validationFunction = common.expectsError({
+    code: 'ERR_INVALID_ARG_TYPE',
+    type: TypeError
+  });
+
+  const testBlockTypeError = (method, block) => {
+    let threw = true;
+
+    try {
+      method(block);
+      threw = false;
+    } catch (e) {
+      validationFunction(e);
+    }
+
+    assert.ok(threw);
+  };
 
-  assert.ok(threw);
+  testBlockTypeError(assert.throws, 'string');
```

Comment:
@addaleax Having a hard time writing this up for Code + Learn in a way that makes it clear why we want to do this. (Probably because I see it as a style preference.) Would you want to try? If not, no biggie, I'll just leave it off the list. We have enough tasks for the next Code + Learn.